test/manipulation.test.js

```diff
@@ -37,7 +37,13 @@ describe('manipulation', function() {
   before(function setupStubUserModel(done) {
     StubUser = db.createModel('StubUser', { password: String }, { forceId: true });
     StubUser.setter.password = function(plain) {
-      this.$password = plain + '-' + plain.toUpperCase();
+      if (plain === null)
+        return;
+      var pos = plain.indexOf('-');
+      // Check that it has not been already hashed
+      if (pos === -1 ||
+      (plain.substr(0, pos).toUpperCase() !== plain.substr(pos + 1, plain.length)))
+        this.$password = plain + '-' + plain.toUpperCase();
```

Comment:
Uff, this is rather difficult to understand. Could you please split the code into smaller steps and save intermediate results into named variables? Something like this:

```js
var hashed = false;
var pos = plain.indexOf('-');
if (pos !== -1) {
  var head = plain.substr(0, pos);
  var tail = plain.substr(pos + 1, plain.length);
  hashed = head.toUpperCase() === tail;
}

if (hashed) return;

this.$password = plain + '-' + plain.toUpperCase();
```